File: SourceCode/Angular/src/app/core/domain-classes/product.ts
import { ProductTax } from './product-tax';
import { Unit } from './unit';
import { UnitConversation } from './unit-conversation';

export interface Product {
  id?: string;
  name: string;
  categoryId: string;
  code?: string;
  barcode?: string;
  skuCode?: string;
  skuName?: string;
  description?: string;
  productUrl?: string;
  unitId: string;
  purchasePrice?: number;
  salesPrice?: number;
  mrp?: number;
  productUrlData?: string;
  isProductImageUpload?: boolean;
  productTaxes?: ProductTax[];
  unit?: UnitConversation;
  categoryName?: string;
  unitName?: string;
  hasVariant?: boolean;
  variantId?: string;
  variantItemId?: string;
  productVariants?: Product[];
  productTaxIds?: string[];
  alertQuantity?: number;
  margin?: number;
  isMarginIncludeTax?: boolean;
  taxAmount?: number;
  brandId?: string;
}

File: SourceCode/SQLAPI/POS.Data/Entities/Product/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using POS.Data.Entities;


namespace POS.Data
{
    public class Product : BaseEntity
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Code { get; set; }
        public string Barcode { get; set; }
        public string SkuCode { get; set; }
        public string SkuName { get; set; }
        public string Description { get; set; }
        public string ProductUrl { get; set; }
        public Guid UnitId { get; set; }
        [ForeignKey("UnitId")]
        public UnitConversation Unit { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal? PurchasePrice { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal Margin { get; set; }
        public bool IsMarginIncludeTax { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal? SalesPrice { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal? Mrp { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal? TaxAmount { get; set; }
        public Guid CategoryId { get; set; }
        [ForeignKey("CategoryId")]
        public ProductCategory ProductCategory { get; set; }
        public Guid? BrandId { get; set; }
        [ForeignKey("BrandId")]
        public Brand Brand { get; set; }
        public bool HasVariant { get; set; }
        public Guid? ParentId { get; set; }
        [ForeignKey("ParentId")]
        public Product ParentProduct { get; set; }
        public Guid? VariantId { get; set; }
        [ForeignKey("VariantId")]
        public Variant Variant { get; set; }
        public Guid? VariantItemId { get; set; }
        [ForeignKey("VariantItemId")]
        public VariantItem VariantItem { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal? AlertQuantity { get; set; }
        public List<ProductTax> ProductTaxes { get; set; }
        public ICollection<Product> ProductVariants { get; set; }
        public List<ProductStock> ProductStocks { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal CurrentStock { get; set; } = 0;

    }
}
